integrationtesting/opendcs-tests/src/test/java/org/opendcs/regression_tests/AlgorithmTestsIT.java

```diff
@@ -303,4 +318,52 @@ private void loadRatingimport(String folderRatingStr)
             }
         }
     }
+    
+    /**
+     * Execute SQL file in the database for test setup
+     * @param sqlFile The SQL file to execute
+     */
+    private void executeSqlFile(File sqlFile)
+    {
+        if (db == null)
+        {
+            log.warn("OpenDcsDatabase not available, skipping SQL file execution: " + sqlFile.getName());
+            return;
+        }
+        
+        try
+        {
+            // Read the SQL file content
+            String sqlContent = new String(Files.readAllBytes(sqlFile.toPath()), StandardCharsets.UTF_8);
+            
+            // Replace office ID placeholder if this is a CWMS database
+            if (tsDb instanceof CwmsTimeSeriesDb)
+            {
+                CwmsTimeSeriesDb cwmsDb = (CwmsTimeSeriesDb) tsDb;
+                String officeId = cwmsDb.getDbOfficeId();
+                sqlContent = sqlContent.replace("DEFAULT_OFFICE", officeId);
+            }
+            
+            // Execute the SQL using a transaction
+            try (DataTransaction tx = db.newTransaction())
+            {
+                Connection conn = tx.connection(Connection.class)
```

Comment:
Needs to use `DAO::inTransaction`.

(Actually this is a single statement so just doing `try (Connection conn = db.getConnection())` and just doing the callable would be sufficient.